examples/dao/assets/dao-fund-lsig.py

```diff
@@ -0,0 +1,54 @@
+import sys
+sys.path.insert(0,'..')
+
+from algobpy.parse import parse_params
+from pyteal import *
+
+def proposal_lsig(ARG_DAO_APP_ID):
+    """
+    Represents DAO treasury (ALGO/ASA)
+    """
+
+    # check no rekeying, close remainder to, asset close to for a txn
+    def basic_checks(txn: Txn): return And(
+        txn.rekey_to() == Global.zero_address(),
+        txn.close_remainder_to() == Global.zero_address(),
+        txn.asset_close_to() == Global.zero_address()
+    )
+
+    # ideally we should restrict the sender here (no one can withdraw funds)
+    receive_or_send = basic_checks(Txn)
+
+    # verify funds are transfered only when paired with DAO app (during execute call)
+    payment = And(
+        # verify first transaction
+        basic_checks(Gtxn[0]),
+        Gtxn[0].type_enum() == TxnType.ApplicationCall,
+        Gtxn[0].application_id() == Int(ARG_DAO_APP_ID),
+        Gtxn[0].application_args[0] == Bytes("execute"),
+
+        # verify second transaction (either payment in asa or ALGO)
+        basic_checks(Gtxn[1]),
+        Or(
+            Gtxn[1].type_enum() == TxnType.AssetTransfer,
+            Gtxn[1].type_enum() == TxnType.Payment,
+        )
+    )
+
+    program = program = Cond(
+        [Global.group_size() == Int(1), receive_or_send],
+        [Global.group_size() == Int(2), payment]
```

Comment:
could you please add a check here for opt-in ? so that malicious user cannot spam proposal lsig with ASA opt-in transactions ?

Reply:
we were thinking about it. There can be many DAO funds. So in essence it is not a problem. So anyone can create a DAO.
Maybe we can restrict it and add a random owner and explain that the owner is only there to control what assets to opt-in, but we can use any fund lsig with DAO.